api/lcp/controller.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import flask
+from flask import Response
+
+from api.controller import CirculationManagerController
+from api.lcp.credential import LCPCredentialFactory
+from api.lcp.factory import LCPServerFactory
+from core.model import Session, ExternalIntegration
+from core.util.problem_detail import ProblemDetail
+
+
+class LCPController(CirculationManagerController):
+    """Contains API endpoints related to LCP workflow"""
+
+    def __init__(self, manager):
+        """Initializes a new instance of LCPController class
+
+        :param manager: CirculationManager object
+        :type manager: CirculationManager
+        """
+        super(LCPController, self).__init__(manager)
+
+        self._logger = logging.getLogger(__name__)
+        self._credential_factory = LCPCredentialFactory()
+        self._lcp_server_factory = LCPServerFactory()
+
+    def _get_patron(self):
+        """Returns a patron associated with the request (if any)
+
+        :return: Patron associated with the request (if any)
+        :rtype: core.model.patron.Patron
+        """
+        self._logger.info('Started fetching an authenticated patron associated with the request')
+
+        patron = self.authenticated_patron_from_request()
+
+        self._logger.info('Finished fetching an authenticated patron associated with the request: {0}'.format(patron))
+
+        return patron
+
+    def _get_lcp_passphrase(self, patron):
+        """Returns a patron's LCP passphrase
+
+        :return: Patron's LCP passphrase
+        :rtype: string
+        """
+        db = Session.object_session(patron)
+
+        self._logger.info('Started fetching a patron\'s LCP passphrase')
+
+        lcp_passphrase = self._credential_factory.get_patron_passphrase(db, patron)
+
+        self._logger.info('Finished fetching a patron\'s LCP passphrase: {0}'.format(lcp_passphrase))
+
+        return lcp_passphrase
+
+    def _get_lcp_collection(self, library):
+        """Returns an LCP collection for a specified library
+        NOTE: We assume that there is only ONE LCP collection per library
+
+        :param library: Library
+        :type library: core.model.library.Library
+
+        :return: LCP collection
+        :rtype: core.model.collection.Collection
+        """
+        lcp_collection = next(iter([
+            collection
+            for collection in library.collections
+            if collection.protocol == ExternalIntegration.LCP
+        ]))
+
+        return lcp_collection
+
+    def get_lcp_passphrase(self):
+        """Returns an LCP passphrase for an authenticated patron"""
+        self._logger.info('Started fetching a patron\'s LCP passphrase')
+
+        patron = self._get_patron()
+        if isinstance(patron, ProblemDetail):
+            return patron
+        if isinstance(patron, Response):
+            return patron
```

Comment:
I see -- you have the equivalent of `requires_auth` inside here. Unless there's a good reason for duplicating the code I'd rather you use `requires_auth` itself.

Reply:
There is no specific reason, I just wasn't aware of `requires_auth`. Added `@requires_auth`